packages/z2s/src/compiler.ts

```diff
@@ -256,46 +266,55 @@ export class Compiler {
           relationship.subquery.table,
         )} ${
           format?.singular ? this.limit(1) : this.limit(lastLimit)
-        } ) ${sql.ident(`inner_${relationship.subquery.alias}`)}
+        } ) ${sql.ident(innerAlias)}
       ) as ${sql.ident(relationship.subquery.alias)}`;
     }
     return sql`(
-      SELECT ${
-        format?.singular ? sql`` : sql`COALESCE(json_agg`
-      }(row_to_json(${sql.ident(`inner_${relationship.subquery.alias}`)})) ${
-        format?.singular ? sql`` : sql`, '[]'::json)`
-      } FROM (${this.select(
+      SELECT ${this.#toJSON(innerAlias, format?.singular)} FROM (${this.select(
         relationship.subquery,
         format,
         this.correlate(
           parentTable,
-          parentTable,
+          parentTableAlias ?? parentTable,
           relationship.correlation.parentField,
           relationship.subquery.table,
           relationship.subquery.table,
           relationship.correlation.childField,
         ),
-      )}) ${sql.ident(`inner_${relationship.subquery.alias}`)}
+      )}) ${sql.ident(innerAlias)}
     ) as ${sql.ident(relationship.subquery.alias)}`;
   }
 
   pullTablesForJunction(
     relationship: CorrelatedSubquery,
-    tables: [string, Correlation, number | undefined][] = [],
-  ) {
+  ): [
```

Comment:
@tantaman This seemed to be trying to support an arbitrary chain of hidden relationships, but afaik we only support a single junction hop.  Was my misunderstanding what this was doing?